src/zenml/zen_stores/migrations/versions/728c6369cfaa_add_name_column_to_input_artifact_pk.py

```diff
@@ -0,0 +1,114 @@
+"""Add name column to input artifact PK [728c6369cfaa].
+
+Revision ID: 728c6369cfaa
+Revises: 0.36.0
+Create Date: 2023-03-20 13:37:51.215760
+
+"""
+from alembic import op
+
+# revision identifiers, used by Alembic.
+revision = "728c6369cfaa"
+down_revision = "0.36.0"
+branch_labels = None
+depends_on = None
+
+
+def upgrade() -> None:
+    """Upgrade database schema and/or data, creating a new revision."""
+    # ### commands auto generated by Alembic - please adjust! ###
+    engine_name = op.get_bind().engine.name
+    if engine_name == "sqlite":
+        constraint_name = "pk_step_run_input_artifact"
+    elif engine_name == "mysql":
+        constraint_name = "PRIMARY"
```

Comment:
Wow this is much easier than for the foreign keys, nice. I assume you tested this works, right?

Reply:
I tested `upgrade -> downgrade -> re-upgrade` on both MySQL and SQLite which both seemed to work (after an hour or so of trying to get this right for both to be fair 😄)